R/plot_inventory_data.R

```diff
@@ -0,0 +1,101 @@
+#' Title
+#'
+#' @param data The data.frame to calculate from
+#' @param station The name of the station column in \code{data}, if the data are for multiple station. 
+#' The summary table is calculated separately for each station.
+#' @param elements The name of the columns in \code{data} to apply the summary calculation to.
+#' @param y Whether to display counts or percentages.
+#' @param columns Whether to display stations or elements on the columns.
+#' @param fill Whether to fill by stations, elements, or neither.
+#' @param facets Whether to facet by stations, elements, or neither
+#' @param display_perc logical. Whether to display percentage values on the plot.
+#' @param display_count logical. Whether to display the count values on the plot.
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' data(daily_niger) 
+#' plot_inventory_data(data = daily_niger, station = "station_name",
+#'                     elements = c("tmin", "tmax"),
+#'                     columns = "stations", facets = "elements")
+#'                     
+#' # can omit station argument
+#' plot_inventory_data(data = daily_niger, 
+#'                     elements = c("tmin", "tmax"))
+#' 
+#' # can view percentage instead
+#' plot_inventory_data(data = daily_niger, y = "percentage",
+#'                     elements = c("tmin", "tmax"), display_perc = TRUE,
+#'                     display_count = TRUE)
+
+plot_inventory_data <- function(data, station = NULL, elements, y = c("count", "percentage"),
+                                columns = c("elements", "stations"),
+                                fill = c("none", "stations", "elements"),
+                                facets = c("stations", "elements", "none"),
+                                display_perc = FALSE, display_count = FALSE){
+  checkmate::assert_logical(display_perc)
```

Comment:
Add `assert_dataframe` for `data`?

Reply:
I’ve done that in ‘ summarise_inventory_data’ which this function calls